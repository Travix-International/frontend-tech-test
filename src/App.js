import React from 'react';
import { 
  Container
} from 'reactstrap';
import { Provider } from 'react-redux';
import configureStore from './store/configureStore';
import HeaderContainer from './containers/HeaderContainer';
import TaskListContainer from './containers/TaskListContainer';
import FooterContainer from './containers/FooterContainer';
import styles from './App.module.scss';

function App() {
  return (
    <Provider store={configureStore()}>
      <Container className={styles['app']}>
        <HeaderContainer className={styles['app-header']} />
        <TaskListContainer />
        <FooterContainer />
    </Container>
    </Provider>
  );
}

export default App;
