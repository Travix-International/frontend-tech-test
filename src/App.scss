@import './styles/common.scss';
@import './styles/header.scss';

.main-container {
    box-sizing: border-box;
    @include for-phone-only {
        nav {
            ul {
                flex-direction: row;
                .nav-link {
                    padding-right: .5rem;
                    padding-left: .5rem;
                }
            }
        }
    }

    .card {
        &.small {
            width: 100%;
            @include for-tablet-landscape-up {
                max-width: 70%;
                margin: auto;
            }
            @include for-desktop-up {
                max-width: 60%;
            }
        }
    }
}