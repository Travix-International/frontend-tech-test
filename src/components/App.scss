@import "../containers/CreateTask.scss";
@import "../containers/List.scss";

.banner {
  background-image: linear-gradient(#0087cd 82%, #22B7BB);
  color: #fff;
  text-align: center;
  overflow: hidden;

  &__logo {
    // Use base64 to ensure loading CRITICAL content
    background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMgAAABSCAMAAAAme2uJAAAAh1BMVEVMaXH///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////8s+2B1AAAALHRSTlMAMJ9g70DPvxCAIN+vcI9QAveUA/4GBQHxBAkxeQ6eCAcL3k0Tqg+LKBIl4mYjT20AAAUbSURBVHja3ZvpgtsmEIBBHAJ0uXX3yNGmV9Kk5f2fr2tb1gADwpZFYnn+rYyAjzkRLNm+fPv64cO/e7Jx+fLxv9fX149k46I0t9bygWxbThhWUrJt6Y4YtlZk0yIae5SKbFpUa0/CyKZltCrLt82hajtybNvNB/4YHL0dRd5RuGJtXVfimjeMvEMOJa+OoILfMccVJJ3NcTy///3p+d1+t3t52e12+/3+3fPT06f3f/xKyklvJ7nQuqoMh9KNTUpbrCLjziCXJcEch0xBFC0ChDNEc5El5uzqtxQBKL6ECHeINTjIL9b+CJUYZwS5BkcepCZFpLliqSg4FDfkzkA6GMFcoI98XfJnDqQjZWSKpuxyDkvnVmZeeLEcqk/9D1dwsEx1n5DSW2I1aO3ERKb1MM/RZfpjVX0U7sz/9EQXy4dIhqP3NwJZ4IL4WWeTBxWBgoxguhOG3CzsPLJOckhyCwhtDzUEPU664oc1G4Cib865WatzsSPZuMLy+Dgsh2pbV2b8o4LmJrHuzMn/ajkIRHDmVHn9iHFAhGHowZrh9z6yiucWnHplPCz8OFaY/88v3QJSTw9k8CMLYwSrnRgpopbSu8WVdruro0WR4S7eLSA0XhKf5oQFVNbGorfxZjVNshqHxtlbun3eBNLFU0uKA7yS+ysZ9qZPawT6af0OFHb020D6BAjDD/1qNrrZaAHEM/+zUwcOz1wd3QQCT4JhqJ0X70U1uTo8056yD78FHqd9BxlICRCuZBakj3jp4HWv4a+YEQ++g5QAkYbZLIiI2FbleEUAgofinoMUAGneFrgJ6nzUyttucLxL1wgEG5ebQdYHqQbPRE6JHicV31QGFMqNDzL+vt4+IgPCNURFCJTwCdD3XlRz9N5WDYMQttrGbh6kUmAiuP6hrk78WMvD7MYwiL9CfnZcGWTyWRMvi7TzOIhR1H+Nh+1hAqsYVgZERKIoT3wWCb4u9H58rZIgqlnDsDIg4dLDlHBhFLRrfFOjGAQZKDT+LiA6sQUKVURd55eoG3f/sYZCbgPBM2tdOx/gj0TzeHXH7gCEgRJcfakUyIi/gm0tBMEzQHHauLuPeRDVFA6/1zg7MpOOQp7PgZyRIecWAnG/ByTCL3rU9vBKBqRa4Sv6ZSDCLbZTCTF8fVJXnwExa+TEy0CIjZUoxitR0tWTSYDMlilDGZA2UDs+9wJ7x4l6HoSuUshfCMK8mdHpsg5Ien3ZLAgMiBy+BMh5neEzcVgiJReYq3kQkTyULQHyts4ZQTEZLHEepE52qUuAvKlkXpK1E50HMe6e/YZapZ0B4dccDjnM/oafzINUrgbkAofHfVO0PQ2RseTNsIufT8dyiEJ6rxeBoCftFbcMZOo8XUUDLU+c6VC+1LgomopwFjNHwuM1RZ2qNab2bcT1TMTBmgVn3x16YvCFvNCG4QgiccXBzJ/qamRG/dKKfpjyT/hE5w5RJZ2qlCoRQvpgKXhg+w2uSeolNRd8aKtVcEbJo12obhq6Oa7qSSc9vtaD+aB5pWaqaiUX5xIhhAnPO9MqNYN+EzY1UELEfE/3vQ46wc1rNOPQ4b/fEfNat2yMs7CunjYhfSJjDJNOCl5dWFOS555UbuqaL/PyJr5W2xW7ErOyeF+ZNyxHV3+Ef0Fw6+mtGFFUmgexLPooluVWhyvcmfpx0ribmQ0L9W7kbFj05irDhMgHCb7GPoiLdI/iIp9/AvmLbFj++Rnkb7Jdedk5srCP/wE/beBpjd86eQAAAABJRU5ErkJggg==");
    width: 200px;
    height: 82px;
    font-size: 0;
    margin: 2rem auto;
    
    &::after {
      content: "tasks";
      font-size: 1rem;
      font-family: 'Varela Round';
      font-weight: 400;
      line-height: 9.4rem;
    }
  }
}

@media (min-width: 1024px) {
  .banner {
    width: 512px;
    float: left;
    // Delegate visual responsability to the body
    background-image: none;
  }
}