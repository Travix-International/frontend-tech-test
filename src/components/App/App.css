main {
  padding-bottom: 4rem; }

.list-title {
  margin-top: 3rem; }
