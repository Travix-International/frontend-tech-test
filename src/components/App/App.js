import React from 'react';

import { Grid, Row, Col } from 'react-flexbox-grid';

import AddTodo from '../../containers/AddTodo';
import VisibleTodoList from '../../containers/VisibleTodoList/';

export default function App() {
  return (
    <Grid id={"appContainer"}>
      <Row>
        <Col
          lg={12}
          md={12}
          sm={12}
          xs={12}
        >
          <AddTodo />
        </Col>
      </Row>
      <Row>
        <Col
          lg={12}
          md={12}
          sm={12}
          xs={12}
        >
          <VisibleTodoList />
        </Col>
      </Row>
    </Grid>
  );
}
