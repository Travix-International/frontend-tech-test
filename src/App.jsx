import React, { Component } from 'react';
import logo from './logo.svg';

//Import the modified App.css
import './App.css';

// Import the Routes component, which contains our Route setup

import { Routes } from './routes';


// Provider component is a react-redux component for setting up the Redux Store

import { Provider } from 'react-redux';

// Import the ConfigureStore that holds the initial Configuration

import { configureStore } from './store/configure-store';

import * as TaskAction from './actions';


// Create a Store from the Configuration, we can pass a Initial State here

const store = configureStore();

// At first dispatch a Get Todos Actions, So we'll recieve the Todos
// fetched from the server at the start of the app

store.dispatch(TaskAction.GetTasks());

const App = (props) => {
  return (

    //Provider needs to contain all the Containers/Components it will give access to the Store

    <Provider store={store} >
      <Routes />
    </Provider>
  )
}

export default App;