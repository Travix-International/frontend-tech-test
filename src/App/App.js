/* eslint-disable */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  BrowserRouter as Router,
} from 'react-router-dom'

import TodoList, { dummyTodos } from 'SRC/TodoList'
import './App.scss'


class App extends Component {

  static propTypes = {
    location: PropTypes.object, // eslint-disable-line
  }

  state = {
    isLoggedIn: false,
  }

  componentDidMount() { }

  handleLogin = () => {
    this.setState({ isLoggedIn: true })
  }

  handleLogout = () => {
    this.setState({ isLoggedIn: false })
  }

  render() {
    // const { isLoggedIn } = this.state
    return (
      <Router>
        <div>
          <h1> Todo List </h1>
          <TodoList todos={[]} />
        </div>
      </Router>
    )
  }
}

export default App
