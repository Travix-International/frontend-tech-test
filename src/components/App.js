import '../styles/App.css';
import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';

// User define components
import Home from './home/homePage';
import About from './about/aboutPage';

class App extends Component {
  render() {
    return (
      <div className="container">
      <BrowserRouter>
          <div>
            <Route exact path="/" component = {Home} />
            <Route path="/about" component = {About} />
          </div>
      </BrowserRouter>
      </div>
    );
  }
}

export default App;
