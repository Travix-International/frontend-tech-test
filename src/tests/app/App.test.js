import React from 'react';
import { shallow } from 'enzyme';
import App from '../../containers/app/App';
import TaskList from '../../containers/app/TasksList';

describe('App Component', () => {
    it('should render correctly in "debug" mode', () => {
      const component = shallow(<App debug />);
    
      expect(component).toMatchSnapshot();
    });
    it('should render nav', () => {
      const component = shallow(<App/>);
    
      expect(component.find('nav')).toBeTruthy();
    });
    it('should render header', () => {
      const component = shallow(<App/>);
      expect(component.find('nav').find('div').text('o Do Lis')).toBeTruthy();
    });
    it('should render TaskList Component', () => {
      const component = shallow(<App/>);
      expect(component.find(TaskList)).toMatchSnapshot();
    });
   
});