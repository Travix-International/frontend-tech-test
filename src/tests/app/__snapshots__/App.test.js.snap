// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render TaskList Component 1`] = `<Connect(TasksList) />`;

exports[`App Component should render correctly in "debug" mode 1`] = `
<div>
  <nav
    className="nav"
  >
    <div
      className="header"
    >
      o Do Lis
    </div>
  </nav>
  <Connect(TasksList) />
</div>
`;
